week1/project/src/components/CategoryList/CategoryList.js

```diff
@@ -0,0 +1,42 @@
+import React, { useState } from "react";
+import CategoryListItem from "./CategoryListItem";
+
+const CategoryList = ({ categories, selectedCategory, onCategoryClick }) => {
+  const [prevSelectedCategory, setPrevSelectedCategory] = useState(null);
+
+  const handleClick = (category) => {
+    if (category === selectedCategory) {
+      onCategoryClick(null);
+      setPrevSelectedCategory(null);
+    } else {
+      onCategoryClick(category);
+      setPrevSelectedCategory(category);
+    }
+  };
+
+  return (
+    <div>
```

Comment:
If this `div` does not have any stylistic reasons to be here (meaning that it needs to be here because of some CSS, like it's a child of a flex parent) then you can also use a "React Fragment". This will make sure you don't render unnecessary `div`s on the page 😉 

```suggestion
    <>your categories code here</>
```

Reply:
Okay, I didn't know about this option of avoiding the <div> if it's not needed for stylistic reasons. I will implement this change.